### DISCUSSION TITLE
PiJuice and Fan Hat

### DISCUSSION THREAD

**Discussion by alpapak** [PiSupply/PiJuice#697] [Q&A] [answered]
I would like to ask if it is possible to have:

- Raspberry Pi 4
- Argon 40 Fan Hat or Waveshare PWM Controlled Fan HAT
- PiJuice Hat

To stack in this order. Will it work or i will have issues?

Links:
[https://www.argon40.com/argon-fan-hat-for-raspberry-pi-4-raspberry-pi-3b-and-raspberry-pi-3-b.html](url)
[https://www.waveshare.com/fan-hat.htm](url)

**Comment by tvoverbeek** [comment, marked as answer]
Electrically it would work (provided the Argon Fan Hat does not use I2C addresses used by the PiJuice).
Hoever I think you will get problems with the airflow since the PiJuice is on top of the fan hat, partially
blocking the airflow.

**Reply by alpapak** [reply]
@tvoverbeek Thank you for your message. From the Argon Fan Hat i got the following answer:

```
ARGON ONE CASE PINS: (same with ARGON 40 FAN HAT)
Power Button - GPIO 11
Shut Down Monitor - GPIO 7 (monitor cut if no signal)
IR Receive (in) - GPIO 18
IR Transmit (out) - GPIO 17
FAN CONTROL -PIN 27 - PIN 28 (i2c)

Fan is controlled via i2c

Here are the codes:
https://github.com/Argon40Tech/Argon-ONE-i2c-Codes
```

All pins are fine, except from i2c(pin 27 - 28). Or i am wrong?

**Reply by tvoverbeek** [reply]
I2C is a serial bus which allows multiple clients as long as they use distinct i2c adresses.
If you check the Argon-ONE-i2c-Codes link you will see the fan uses address 0x1A which does not conflict
wiht the PiJuice ( addresses 0x14 and 0x68)
Also it seems the fan hat is on the normal i2c-1 bus (pins 3 and 5).
i2c-0 bus is on pins 27 and 28 and is used to read the EEPROM from a HAT.
If the Argon HAT also has an EEPROM for automatic GPIO configuration  you will have to change the PiJuice EEPROM address to 0x52 to avoid a conflict.
conflicts.

**Reply by alpapak** [reply]
@tvoverbeek Thank you very much for your help and knowledge. I have an extra pi4, waiting for pijuice to order and i will test it.

**Comment by mbstechnologiessales** [comment]
A little late in responding to this, but may help others in the future. I currently have this setup. I have the Argon Fan Hat plugged into the RPI directly so it can cool the CPU, and then used a Male/Male GPIO header to attach the PiJuice on top of the Fan Hat. I used two different sized standoffs to get the correct height and it still allows plenty of airflow. I did run into problems with the fan hat using its internal microcontroller for the power button. It experienced unpredictable rebooting and fan speed changes. Changing the eeprom address may have fixed these issues, but I didn't need the power button controls as I will not have physical access to the system anyway. The microcontroller for the power button is only active if the script is installed. I removed the script and used a replacement argononed daemon made by DarkElvenAngel and configured it to only allow controlling fan speed, and not the power button before installing. Otherwise, no conflicts and everything has stayed cool and working great.
